proposals/collection-expression-arguments.md

```diff
@@ -127,6 +127,20 @@ collection_element
 +  ;
 ```
 
+There is a syntactic ambiguity immediately introduced with this grammar production.  Similar to the ambiguity
+between `spread_element` and `expression_element` (explained [here](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md#detailed-design), there is an immediately syntactic ambiguity between `with_element` and `expression_element`.
+Specifically `with(<arguments>)` is both exactly the production-body for `with_element`, and is also reachable through
+`expression_element -> expression -> ... -> invocation_expression`. There is a simple overarching rule for
+collection_elements. Specifically, if the element [lexically](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/lexical-structure.md)
+starts with `with(` then it is always treated as a `with_element`.
```

Comment:
What's our intention for the value of a KVP? Still a `with()` expression?

Reply:
actually, you said `with() expression` not `with-element`.  SO in that case, YES.  it would be an 'expression'. not an element.